pubsub/tests/unit/pubsub_v1/publisher/batch/test_thread.py

```diff
@@ -217,21 +230,35 @@ def test_publish():
     )
 
     # Publish each of the messages, which should save them to the batch.
-    for message in messages:
-        batch.publish(message)
+    futures = [batch.publish(message) for message in messages]
 
     # There should be three messages on the batch, and three futures.
     assert len(batch.messages) == 3
-    assert len(batch._futures) == 3
+    assert batch._futures == futures
 
     # The size should have been incremented by the sum of the size of the
     # messages.
-    assert batch.size == sum([m.ByteSize() for m in messages])
+    expected_size = sum([message_pb.ByteSize() for message_pb in messages])
+    assert batch.size == expected_size
     assert batch.size > 0  # I do not always trust protobuf.
 
 
-def test_publish_max_messages():
-    batch = create_batch(max_messages=4)
+def test_publish_not_will_accept():
```

Comment:
@lukesneeringer Note that I went with `not_will_accept` over `will_not_accept` here to make `git grep will_accept` slightly more useful. I can change to better grammar if you see fit.

Reply:
That seems like a reasonable decision.